components/PreviewFeaturesProvider.js

```diff
@@ -0,0 +1,38 @@
+import React, { useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const PreviewFeaturesContext = React.createContext({
+  setShowPreviewFeatures: () => {},
+});
+
+const PreviewFeaturesProvider = ({ children }) => {
+  const [showPreviewFeatures, setShowPreviewFeatures] = useState(false);
+
+  const context = {
+    showPreviewFeatures,
+    setShowPreviewFeatures,
+  };
+
+  return <PreviewFeaturesContext.Provider value={context}>{children}</PreviewFeaturesContext.Provider>;
```

Comment:
Avoid creating dynamic objects that expose the value of a context like this, it'll cause unwanted re-renders even if none of the value changes but the parent of the provider re-renders (the old React docs had a section about it https://legacy.reactjs.org/docs/context.html#caveats).

One way to avoid it is to memorize it:

```suggestion
  const context = useMemo(() => ({
    showPreviewFeatures,
    setShowPreviewFeatures,
  }), [showPreviewFeatures, setShowPreviewFeatures]);

  return <PreviewFeaturesContext.Provider value={context}>{children}</PreviewFeaturesContext.Provider>;
```